plugins/modules/netconf_config.py

```diff
@@ -56,13 +59,18 @@
   format:
     description:
     - The format of the configuration provided as value of C(content). Accepted values
-      are I(xml) and I(text) and the given configuration format should be supported
-      by remote Netconf server.
+      are I(xml), I(text) and I(json).
+    - In case of json string format it will be converted to the corresponding xml string using
+      xmltodict library before pushing onto the remote host.
+    - In case of I(text) format of the configuration should be supported by remote Netconf server.
+    - If the value of C(format) options is not given it tries to guess the data format of
+      C(content) option as one of I(xml) or I(json) or I(xpath).
```

Comment:
Do you mean text here? I haven't seen `xpath` mentioned anywhere prior.
```suggestion
      C(content) option as one of I(xml) or I(json) or I(text).
```